src/util/positioning.ts

```diff
@@ -89,6 +93,42 @@ export class Positioning {
     return elOffset;
   }
 
+  private _getScrolledParent(element, includeHidden = false) {
+    let style = getComputedStyle(element);
+    const excludeStaticParent = style.position === 'absolute';
+    const overflowRegex = includeHidden ? /(auto|scroll|hidden)/ : /(auto|scroll)/;
+
+    if (style.position !== 'fixed') {
+      for (let parent = element; parent = parent.parentElement;) {
+        style = getComputedStyle(parent);
+        if (excludeStaticParent && style.position === 'static') {
+          continue;
+        }
+        if (overflowRegex.test(style.overflow + style.overflowY + style.overflowX)) {
+          return parent;
+        }
+      }
+    }
+
+    return null;
+  }
+
+  subscribeScroll(hostElement: HTMLElement, callback: Function): Function {
```

Comment:
Could you refactor it to be like to the `ngbAutoClose` and `ngbFocusTrap` we have ?

```ts
// here is a sketch:

export function ngbScroll(hostElement: HTMLElement, stopped$: Observable<any>, callback: Function) {

   runOutsideAngular(() => {

        // to be used like
        fromEvent(...).pipe(takeUntil($stopped));

   });

}
```

- rename to `ngbScrollProvider` or something similar
- move `runOutsideAngular` inside the function
- pass `stop$` observable to avoid unsubscribing